### ISSUE TITLE
[Bug]: Imported m3u playlists not preserving order of the file

### ISSUE THREAD

**Issue by charliec111** [navidrome/navidrome#3307]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.53.2

### Current Behavior

When an m3u playlist is imported, the order in Navidrome does not reflect the m3u file’s order.

### Expected Behavior

Navidrome playlist order should reflect the m3u file.

### Environment

```markdown
- OS: Debian (Bookworm)
- Browser: Chrome on Linux
- Client: play:sub
```

### How Navidrome is installed?

Docker

### Anything else?

An example playlist made in Navidrome:
![IMG_8182](https://github.com/user-attachments/assets/43e4026b-38f5-4028-8867-7b5f2678a74c)
After exporting from Navidrome and fixing the paths, the m3u file looks like:

```
#EXTM3U
#PLAYLIST:test playlist
#EXTINF:261,Taylor Swift - Down Bad
../Taylor Swift/THE TORTURED POETS DEPARTMENT/04 Down Bad.flac
#EXTINF:233,HAIM - My Song 5
../HAIM/Days Are Gone/01-08 My Song 5.flac
#EXTINF:340,Taylor Swift - But Daddy I Love Him
../Taylor Swift/THE TORTURED POETS DEPARTMENT/06 But Daddy I Love Him.flac
#EXTINF:185,Chappell Roan - HOT TO GO!
../Chappell Roan/The Rise and Fall of a Midwest Princess/07 HOT TO GO!.flac
#EXTINF:218,Michelle Branch - All You Wanted
../Michelle Branch/The Spirit Room/03 All You Wanted.flac
#EXTINF:254,Taylor Swift - Guilty as Sin?
../Taylor Swift/THE TORTURED POETS DEPARTMENT/09 Guilty as Sin_.flac
#EXTINF:246,HAIM - The Wire
../HAIM/Days Are Gone/01-03 The Wire.flac
```

In Navidrome’s import of that m3u file, the order does not match the m3u file it came from:
![IMG_8183](https://github.com/user-attachments/assets/6834c06b-19d5-44de-945d-5363e20814aa)

I have observed Navidrome groups albums together so I suspect there is some sorting going on.

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by deluan** [comment]
Thanks for reporting. Yes, I can confirm this was caused by the optimization in 0.53.2: https://github.com/navidrome/navidrome/commit/46be041e7b86f3fe2bc6cf5f1f64a80c757c0f75

**Comment by daniele-athome** [comment]
Can we downgrade for the moment? Are there any database migrations between 0.53.1 and 0.53.2?

**Comment by deluan** [comment]
Yes, it is safe to downgrade to 0.53.1. A fix for this will be in place for the weekend and available in `develop`, but a new proper release may be a month or so away.

**Comment by deluan** [comment]
This is fixed in #3307 by @kgarner7. Can someone please verify, by running the version of that branch? Binary is attached to a comment in the PR. Thanks!
